### PR TITLE
added api for `cupy.float_power`

### PR DESCRIPTION
**Pull request by amanchhaparia**
Hi,
This PR **aims** to resovle `cupy.float_power` as requested in #6078.

I am getting the data_type assertion error for complex datatype. 4 testcases are being failed.
expect data_type is `numpy.complex128` whereas output generated is `cupy.complex64`.

Looking for your guidance! Thank You.

### CODE REVIEW

File: cupy/_math/arithmetic.py [hunk 1]
```diff
@@ -111,6 +111,17 @@ def imag(val):
 
 floor_divide = _core.floor_divide
 
+float_power = _core.create_ufunc(
+    'cupy_float_power',
+    (('dd->d'),
+     ('DD->D', 'out0 = in1 == in1_type(0) ? in1_type(1): pow(in0, in1)')),
```
Comment (emcastillo) [inline cupy/_math/arithmetic.py:117]: you are missing the `F` dtype (complex64)

File: tests/cupy_tests/math_tests/test_arithmetic.py [hunk 2]
```diff
@@ -228,7 +228,7 @@ def check_binary(self, xp):
         # TODO(niboshi): Fix this. If rhs is a Python complex,
         #    numpy returns complex64
         #    cupy returns complex128
-        if xp is cupy and isinstance(arg2, complex):
+        if xp is cupy and isinstance(arg2, complex) and self.name != 'float_power':
```
Comment (amanchhaparia) [inline tests/cupy_tests/math_tests/test_arithmetic.py:231]: I added the missing `F` datatype but still error persisted,
Actually I had tested for `float_power` for the test cases available for `cupy.power`, I found that here dtype was **converting** to **complex64** which was causing the error. 
@emcastillo, kindly can you please review  and provide me with further guidance.

File: tests/cupy_tests/math_tests/test_arithmetic.py [hunk 3]
```diff
@@ -228,7 +228,8 @@ def check_binary(self, xp):
         # TODO(niboshi): Fix this. If rhs is a Python complex,
         #    numpy returns complex64
         #    cupy returns complex128
-        if xp is cupy and isinstance(arg2, complex):
+        if (xp is cupy and isinstance(arg2, complex)
+                and self.name != 'float_power'):
```
Comment (amanchhaparia) [inline tests/cupy_tests/math_tests/test_arithmetic.py:232]: I added the missing `F` datatype but still error persisted,
Actually I had tested for `float_power` for the test cases available for `cupy.power`, I found that here dtype was **converting** to **complex64** which was causing the error. 
@emcastillo, kindly can you please review  and provide me with further guidance.

Comment (emcastillo) [reply tests/cupy_tests/math_tests/test_arithmetic.py:232]: in `float_power` everything gets promoted to `float64` but complex numbers, which gets promoted to `complex128`

Comment (emcastillo) [reply tests/cupy_tests/math_tests/test_arithmetic.py:232]: tests passed in my environment, Let me check the CIs

**Review by emcastillo — APPROVED** [review]

**Pull request merged**